scanner/engine/worker.cpp

```diff
@@ -1501,6 +1501,9 @@ bool WorkerImpl::process_job(const proto::BulkJobParameters* job_params,
   i32 next_gpu_idx = 0;
   std::mutex startup_lock;
   std::condition_variable startup_cv;
+  i32 resources_fetched_count = 0;
```

Comment:
Maybe this should be `std::atomic<i32>`? I'm not sure if it makes a huge difference here since it''s already behind a lock.

Reply:
Yeah I think it's ok because we're guarded by a lock.

Although ideally we could do the Rust thing of having a Mutex<i32> ...